app/controllers/coronavirus_form/product_details_controller.rb

```diff
@@ -5,24 +5,25 @@ class CoronavirusForm::ProductDetailsController < ApplicationController
   TEXT_FIELDS = %w(product_name product_quantity product_cost certification_details product_postcode product_url lead_time).freeze
 
   def show
-    session["product_details"] ||= []
-    @product = find_product(params["product_id"], session["product_details"])
+    session[:product_details] ||= []
+    @product = find_product(params["product_id"], session[:product_details])
     super
   end
 
   def submit
-    @product = current_product(params[:product_id], session[:product_details]).merge(sanitized_product(params))
-    add_product_to_session(@product)
+    @product = current_product(params["product_id"], session[:product_details]).merge(sanitized_product(params))
```

Comment:
Does the session need to be updated here?

Reply:
The session isn't being updated here.  Because of the nature of products (and the user being able to add an unspecified number of them), we have a `ProductHelper` which has a `add_product_to_session` method that deals with sessions being updated.

Reply:
Ah, I guess I was confused because the session is being used to find the product.